main.go

```diff
@@ -276,4 +303,35 @@ func main() {
 		err = foundation.RunCommandWithArgsExtended(ctx, "kubectl", []string{"rollout", "status", "daemonsets", ds, "-n", params.Namespace})
 	}
 
+	if params.JobTimeoutSeconds > 0 {
```

Comment:
You're better of using a select statement for timeout, see https://gobyexample.com/timeouts

Reply:
Would look something like:

```go
jobsDone := make(chan bool, 1)

go func() {
    for _, job := range params.Jobs {
        log.Info().Msgf("Waiting for job '%v' to finish...", job)
        for true {
            out, _ := foundation.GetCommandWithArgsOutput(ctx, "kubectl", []string{"get", "job", job, "-n", params.Namespace, "-o", "jsonpath='{.status.succeeded}'"})
            if strings.Compare(out, "'1'") == 0 {
                log.Info().Msgf("Job '%v' finished successfully.", job)
                break
            } else {
                time.Sleep(time.Second * 2)
            }
        }
    }
    jobsDone <- true
}()

select {
case res := <-jobsDone:
    log.Info().Msg("All jobs finished successfully.")
case <-time.After(time.Second * time.Duration(params.JobTimeoutSeconds)):
    log.Fatal().Msgf("Job(s) failed to complete successfully within timeout %d seconds.", params.JobTimeoutSeconds)
}
```